### ISSUE TITLE
Style/ParallelAssignment: swap detection for ActiveRecord

### ISSUE THREAD

**Issue by sandfoxme** [bbatsov/rubocop#3286]
**Summary:** The Style/ParallelAssignment should ignore `self.` on the left when detecting variable swap

**Actual behavior:**

I have a code like

```ruby
self.width, self.height = self.height, self.width
```

It says:

```
C: Redundant self detected.
        self.width, self.height = self.height, self.width
                                  ^^^^^^^^^^^
C: Redundant self detected.
        self.width, self.height = self.height, self.width
                                               ^^^^^^^^^^
```

That is expected, ok. I fix it

```ruby
self.width, self.height = height, width
```

Now I get

```
C: Do not use parallel assignment.
        self.width, self.height = height, width
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Expected behavior**: No Style/ParallelAssignment vioaltions should be found in the line

I guess the cop should treat vars on the left with `self.` like vars on the right without `self.`

```
$ rubocop -V
0.41.1 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-linux)
```

**Comment by bbatsov** [comment]
Makes sense.
